airflow/models.py

```diff
@@ -4037,11 +4076,27 @@ def get_many(
 
         query = (
             session.query(cls)
-            .filter(and_(*filters))
-            .order_by(cls.execution_date.desc(), cls.timestamp.desc())
-            .limit(limit))
-
-        return query.all()
+                .filter(and_(*filters))
+                .order_by(cls.execution_date.desc(), cls.timestamp.desc())
+                .limit(limit))
+        results = query.all()
+        if results:
```

Comment:
I'm just going to remove this if, it's not needed (it was added in the original PR).